src/gui/qgsrasterlayersaveasdialog.cpp

```diff
@@ -845,6 +907,32 @@ bool QgsRasterLayerSaveAsDialog::validate() const
   return true;
 }
 
+bool QgsRasterLayerSaveAsDialog::outputLayerExistsInGpkg() const
+{
+  QgsRasterLayer *layer = nullptr;
+  layer = new QgsRasterLayer( QStringLiteral( "GPKG:%1:%2" ).arg( outputFileName(), outputLayerName() ), "", QStringLiteral( "gdal" ) );
```

Comment:
this leaks layer. Better use a unique_ptr here.

Reply:
Ooops, sorry. Now I also made this function more generic  - not only for GPKG. Currently not used for other formats, but much better readable.